### PR TITLE
[Onnx clip]Adding the clip_onnx to our avaible models for faster inference

### PR DESCRIPTION
**Pull request by wanliAlex**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
features

* **What is the current behavior?** (You can also link to an open issue here)
no onnx model for clip

* **What is the new behavior (if this is a feature change)?**
we have onnx model for clip model

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
no

* **Have unit tests been run against this PR?** (Has there also been any additional testing?)
not yet

* **Related Python client changes** (link commit/PR here)
no

* **Related documentation changes** (link commit/PR here)
not yet 

* **Other information**:
the model passes the performance test on a 50k image dataset
the detailed performance can be checked at [link](https://github.com/wanliAlex/fast-clip-research)
check this [site](https://huggingface.co/Marqo) for available models on huggingface 
check this [repo](https://github.com/marqo-ai/onnx-clip-converter) for onnx model generation

* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: src/marqo/s2_inference/onnx_clip_utils.py [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+import open_clip
+from huggingface_hub import hf_hub_download
+from marqo.s2_inference.types import *
+from marqo.s2_inference.logger import get_logger
+import onnxruntime as ort
+
+logger = get_logger(__name__)
+
+_HF_MODE_DOWNLOAD = {
+    '''
```
Comment (pandu-k) [inline src/marqo/s2_inference/onnx_clip_utils.py]: Typo? Should it be `_HF_MODEL_DOWNLOAD `?

**Review by pandu-k — CHANGES_REQUESTED** [review]
The main piece of feedback is that onnx_clip_utils seems to be an outdated copy of clip_utils. Can you make ONNX_CLIP a subclass of CLIP? Having two copies of the same code isn't very [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) - unless you want to decouple the APIs for these two types of models.

We can already see issues with copying the code: clip_utils has already had an update with better error message, which hasn't propagated to your code: https://github.com/marqo-ai/marqo/blob/mainline/src/marqo/s2_inference/clip_utils.py#L62

**Comment by pandu-k** [comment]
Thanks for your work on this PR @wanliAlex !

Just have a question:

Has this feature (not just the models in isolation) been tested at scale? We need to ensure that indexing and retrieval works as expected as part of the Marqo application.

**Comment by wanliAlex** [comment]
> 

Yes, the end to end __indexing__ and __searching__ are tested on a 50k image dataset.

**Review by pandu-k — APPROVED** [review]
Thanks for this PR @wanliAlex !

**Pull request merged**